### PR TITLE
My Blog App : Processing data in models

### PR DESCRIPTION
**Pull request by ohyajk**
In this branch, I have made the following changes:-

- [x]  Created model classes for all entities.
- [x]  Set up associations between models.
- [x]  Created 2 users
- [x]  Created 4 posts written by one of the users.
- [x]  Created 6 posts comments for one of the posts created
- [x]  In the User model, I created a method that returns the 3 most recent posts for a given user.
- [x]  In the Post model, I created a method that updates a user's post count.
- [x]  In the Post model, I created a method that returns the 5 most recent comments for a given post.
- [x]  In the Comment model, I created a method that updates the comments counter for a post.
- [x]  In the Like model, I created a method that updates the likes counter for a post.

Please review my changes
Thanks🍻

### CODE REVIEW

File: app/models/post.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  has_many :comments
+  has_many :likes
+
+  def update_users_posts_counter
+    user.increment(:posts_counter)
```
Comment (gemmen29) [inline app/models/post.rb]: - [x] Kindly fix this typo error as the function is called `increment!`, not `increment`. The function `increment` has different usage and parameters.

File: app/models/user.rb [hunk 2]
```diff
@@ -0,0 +1,9 @@
+class User < ApplicationRecord
+  has_many :posts, foreign_key: 'author_id'
+  has_many :comments, foreign_key: 'author_id'
+  has_many :likes, foreign_key: 'author_id'
+
+  def most_recent_posts
+    posts.oreder(created_at: :desc).limit(3)
+  end
```
Comment (gemmen29) [inline app/models/user.rb]: - [x] Kindly fix this typo error as the function is called `order`, not `oreder`.

File: app/models/post.rb [hunk 3]
```diff
@@ -0,0 +1,13 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  has_many :comments
+  has_many :likes
+
+  def update_users_posts_counter
```
Comment (gemmen29) [inline app/models/post.rb:6]: - [ ] [OPTIONAL] Please use the callback `after_save` to call the `update_users_posts_counter`, so the method is called automatically every time a new post has been saved to the database, then make the `update_users_posts_counter` method private.

File: app/models/like.rb [hunk 4]
```diff
@@ -0,0 +1,8 @@
+class Like < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  belongs_to :post
+
+  def update_posts_likes_counter
+    post.increment!(:likes_counter)
+  end
```
Comment (gemmen29) [inline app/models/like.rb:7]: - [ ] [OPTIONAL] Please use the callback `after_save` to call the `update_users_likes_counter`, so the method is called automatically every time a new post has been saved to the database, then make the `update_likes_posts_counter` method private.

File: app/models/comment.rb [hunk 5]
```diff
@@ -0,0 +1,8 @@
+class Comment < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  belongs_to :post
+
+  def update_comments_posts_counter
+    post.increment!(:comments_counter)
+  end
```
Comment (gemmen29) [inline app/models/comment.rb:7]: - [ ] [OPTIONAL] Please use the callback `after_save` to call the `update_users_comments_counter`, so the method is called automatically every time a new post has been saved to the database, then make the `update_users_comments_counter` method private.

**Review by gemmen29 — CHANGES_REQUESTED** [review]
Hi @ohyajk, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights 💯 
- No linters errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 

## Required Changes ♻️
_Check the comments under the review._



## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by zieeco — APPROVED** [review]
# status: APPROVED :green_circle: :100: 

Hi, @ohyajk, 





Good job so far 👍 Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


## Highlights ⚡ 

✅ Descriptive PR 👍 
✅ Nice data models 🔥 
✅ Professional README file 💯 
✅ Good commit messages 👍 
✅ Linters are passing 👏 


## Optional suggestions
*Check the comment by the previous reviewer*

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**